jina/executors/indexers/keyvalue.py

```diff
@@ -78,22 +79,17 @@ def query(self, key: int) -> Optional[bytes]:
                 return m[r:]
 
     def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
-        # check: hard fail (raises) on key not found
-        missed = []
-        for key in keys:
-            if self.query_handler.header.get(key) is None:
-                missed.append(key)
-        if missed:
-            raise KeyError(f'Key(s) {missed} were not found in {self.save_abspath}')
-
-        # hack
-        self.query_handler.close()
-        self.handler_mutex = False
-        self.delete(keys)
+        keys = check_keys_exist(keys, self.query_handler.header.keys(), self.save_abspath, self.logger)
```

Comment:
if check does return something the name seems wrong. Also try to keep variables inmutable as much as possible

Reply:
Yes, good point, renamed